### DISCUSSION TITLE
how to find and flash the stock firmware?

### DISCUSSION THREAD

**Discussion by Shimakuchi** [RavenSystem/esp-homekit-devices#1583] [Q&A]
Hi, can anyone please provide me a "Revert to Stock Firmware for dummies" guide. Where can I find the stock firmware for the 2.5. I thing I already found some but first I don't know if this is a "clean" file maybe some of you know a official source  🤷🏻‍♂️ and second there are some bin files in the zip. Which one do I need to hand over to the esptool? thanks in advance

--
just to let you know. I run 11 shelly 2.5 und some shelly 1 all with the ravensystem firmware. The all work fine and i think it is the best firmware out there. thats why i preflashed all devices even the one i have spare 😅 but now i need to test something with the stock firmware. thats why i need to find a way back.

**Comment by RavenSystem** [comment]
Each firmware has a different way to flash and install. Different brands use different SDKs that provide binary files in different formats, then there is not an unified way to flash a device. Best way will be to ask to official support of your device, or try to find a standard binary file to flash with esptool.

**Comment by mariuszbeltowski** [comment]
I am looking for the same - The guide of how to flash shelly 2.5 with the original stock firmware, on which I have HAA actually.
